app/src/main/java/de/westnordost/streetcomplete/data/user/statistics/StatisticsDownloader.kt

```diff
@@ -1,31 +1,30 @@
 package de.westnordost.streetcomplete.data.user.statistics
 
 import de.westnordost.streetcomplete.ApplicationConstants
-import java.io.IOException
-import java.net.HttpURLConnection
-import java.net.URL
+import io.ktor.client.HttpClient
+import io.ktor.client.call.body
+import io.ktor.client.request.get
+import io.ktor.client.request.header
+import io.ktor.http.HttpStatusCode
 
 /** Downloads statistics from the backend */
 class StatisticsDownloader(
+    private val httpClient: HttpClient,
     private val baseUrl: String,
     private val statisticsParser: StatisticsParser
 ) {
-    fun download(osmUserId: Long): Statistics {
-        (URL("$baseUrl?user_id=$osmUserId").openConnection() as HttpURLConnection).run {
-            useCaches = false
-            doOutput = true
-            doInput = true
-            setRequestProperty("User-Agent", ApplicationConstants.USER_AGENT)
-            requestMethod = "GET"
-            when (responseCode) {
-                HttpURLConnection.HTTP_OK -> {
-                    return statisticsParser.parse(inputStream.bufferedReader().use { it.readText() })
-                }
-                else -> {
-                    val errorMessage = responseMessage
-                    val errorDescription = errorStream?.bufferedReader()?.use { it.readText() }
-                    throw IOException("$responseCode $errorMessage: $errorDescription")
-                }
+    suspend fun download(osmUserId: Long): Statistics {
+        val response = httpClient.get("$baseUrl?user_id=$osmUserId") {
+            header("User-Agent", ApplicationConstants.USER_AGENT)
+        }
+        when (response.status) {
+            HttpStatusCode.OK -> {
+                return statisticsParser.parse(response.body())
+            }
+            else -> {
+                val errorMessage = response.status.description
+                val errorDescription = response.body<String>()
+                throw Exception("${response.status.value} $errorMessage: $errorDescription")
```

Comment:
I know you just copied the behavior from before, but maybe it is possible to tell Ktor to just throw an exception by itself instead of creating it oneself from the information extracted prior from `HttpURLConnection`?